eis_toolkit/spatial_analyses/cell_based_association.py

```diff
@@ -0,0 +1,573 @@
+            values can be imputed for each added GeoDataFrame(s).
+        add_name: Name of the column(s) to add to the matrix.
+        add_buffer: Allow the use of a buffer around shapes before the
+            intersection with CBA cells for the added GeoDataFrame(s). Minimize
+            border effects or allow increasing positive samples (i.e. cells
+            with mineralization). The size of the buffer is computed using the
+            CRS (if projected CRS in meters: value in meters).
+
+    Returns:
+        CBA matrix is created.
+    """
+
+    for i, data in enumerate(geodata):
+        if i == 0:
+            # Initialization of the CBA matrix
+            grid, cba = _init_from_vector_data(cell_size, geodata[0], column[0], subset_target_attribute_values[0])
+        else:
+            # If necessary, adding data to matrix
+            cba = _add_layer(
+                cba,
+                grid,
+                geodata[i],
+                column[i],
+                subset_target_attribute_values[i],
+                add_name[i - 1],
+                add_buffer[i - 1],
+            )
+
+    _to_raster(cba, output_path)
+
+    return cba
+
+
+@beartype
+def _init_from_vector_data(
+    cell_size: int,
+    geodataframe: gpd.GeoDataFrame,
+    column: str = "",
+    subset_target_attribute_values: Optional[list] = None,
+) -> Tuple[gpd.GeoDataFrame, gpd.GeoDataFrame]:
+    """Creation of CBA matrix.
+
+    Initializes a CBA matrix from a vector file. The mesh is calculated
+    according to the geometries contained in this file and the size
+    of cells.
+
+    Args:
+        cell_size: Size of the cells.
+        geodataframe: GeoDataframe to create the CBA matrix.
+        column: Name of the column of interest.
+            If no attribute is specified, then an artificial attribute is
+            created representing the presence or absence of the geometries of
+            this file for each cell of the CBA grid. If a target attribute is
+            indicated, it must be categorical. Other types of attributes
+            (numerical/string) are not managed. A categorical attribute will
+            generate as many columns (binary) in the CBA matrix than values
+            considered of interest(dummification). See parameter
+            <subset_target_attribute_values>
+        subset_target_attribute_values: List of values of interest of the
+            target column, in case a categorical target attribute has been
+            specified. Allows to filter a subset of relevant values.
+
+    Returns:
+        Tuple of GeodataFrames: the grid mesh produced and the CBA matrix.
+    """
+
+    # Allows a non-mutable default value
+    if subset_target_attribute_values is None:
+        subset_target_attribute_values = []
+
+    # Reading the vector file
+    geodata = geodataframe.copy(deep=True)
+
+    # Initialization of the grid
+    grid = _get_grid(geodata, cell_size, cell_size)
+    grid.crs = geodata.crs
```

Comment:
In _get_grid, the crs seems to be set to this same crs already, so this line could be removed for clarity